examples/twittermap/web/public/javascripts/map/countmap.js

```diff
@@ -0,0 +1,476 @@
+angular.module('cloudberry.map')
+  .controller('countMapCtrl', function($scope, $rootScope, $window, $http, $compile, cloudberry, leafletData, cloudberryConfig, Cache) {
+    // map type change handler
+    // initialize the map (styles, zoom/drag handler, etc) when switch to this map
+    // clear the map when switch to other map
+    $rootScope.$on('maptypeChange', function (event, data) {
```

Comment:
For this remember old maptype logic, I have some doubts that it looks not clean and elegant.
I think a better way is to let the event data tell the listener this maptype change is from which type changed to which type:
We could modify the sidebar/controller.js to include both pre-maptype and post-maptype into the data of this event.
```
if (cloudberry.parameters.maptype !== 'countmap') {
                var premaptype = cloudberry.parameters.maptype; // * Add one line here * //
                cloudberry.parameters.maptype = 'countmap';
                icon1.src = "/assets/images/aggregation_map.png";
                icon2.src = "/assets/images/heat_map_no_border.png";
                icon3.src = "/assets/images/point_map_no_border.png";
                $rootScope.$emit("maptypeChange", [premaptype, cloudberry.parameters.maptype]); // * Modify this line to give an [] as event data * //
}
```